Bio/SeqAnalysis/SeqDatabase.py

```diff
@@ -0,0 +1,277 @@
+        """
+        In several steps, __init__ method is downloading and saving multiple fasta files in a short period of time:
+        1) Checks if the parameters are of the correct type, if not, raises exceptions
+        2) Checks if directories needed for proper file organization exists, if not, creates them
+        3) Prepares list of urls based on the list of IDs from which files will be obtained
+        4) Downloads and saves files using multiprocessing
+        5) Saves database in one fasta file ready for parse by SeqIO module. Also saves additional outputs which are
+           text files that contains entries not downloaded either because of being obsolete or due to http errors
+
+        :param full_list: Either List of IDs or single str.
+        :param directory: str given by user to organize files properly.
+
+        :raises: Exception when variables are not of the expected type.
+
+        """
+        if isinstance(full_list, str):
+            clear_ids = [full_list]
+        elif isinstance(full_list, List):
+            clear_ids = full_list
+        else:
+            raise Exception("Given variable is not either str or List.")
+        if isinstance(directory, str):
+            analysis_dir = directory
+        else:
+            raise Exception("Your directory name is not str.")
+
+        clear_ids = list(dict.fromkeys(clear_ids))
+
+        for list_element in clear_ids:
+            if isinstance(list_element, str):
+                pass
+            else:
+                raise Exception("At least one element of your IDs list is not str.")
+
+        print("Number of unique IDs: {0}".format(len(clear_ids)))
+        print("Name of the current batch of files: {0}".format(analysis_dir))
+
+        print("------------------------------------Directory check------------------------------------------")
+
+        dirs = ['Downloads', 'Failed', 'Database']
+        for single_dir in dirs:
+            self.mk_dirs(single_dir)
+        self.mk_subdirs(analysis_dir)
+
+        obsolete = []
+        http_error = []
+
+        print("--------------------------------------Downloading--------------------------------------------")
+
+        urls = self.paths_and_urls(clear_ids, analysis_dir)
+
+        print("Downloading files. Please wait...")
+
+        ThreadPool(20).imap_unordered(self.fetch_url, urls)
+        start = timer()
+        for entry in urls:
+            self.fetch_url(entry, obsolete, http_error)
+
+        Entrez.email = ''
+        for idx in clear_ids:
+            if re.match(r'([A-Z]){2,3}\w+', idx):
+                self.download_ncbi(idx, analysis_dir, obsolete, http_error)
+
+        obsolete = list(dict.fromkeys(obsolete))
+
+        mypath = ('./Downloads/{0}'.format(analysis_dir))
+        only_files = [f for f in os.listdir(mypath) if isfile(join(mypath, f))]
+        files_no_ext = [".".join(f.split(".")[:-1]) for f in os.listdir(mypath) if os.path.isfile(join(mypath, f))]
```

Comment:
Try to not use generic names such as `f`, `i` etc. Also, lines like this:
`files_no_ext = [".".join(f.split(".")[:-1]) for f in os.listdir(mypath) if os.path.isfile(join(mypath, f))]`
are hard to read and some explanation would be great.

Keep in mind that with complex list/dict comprehensions (with more iterations/conditions), it's more readable to write them in longer form such as:
```
files_no_ext = [
    ".".join(f.split(".")[:-1])
    for f in os.listdir(mypath)
    if os.path.isfile(join(mypath, f))
]
```